active-rfcs/0000-build-performance.md

```diff
@@ -0,0 +1,187 @@
+- Start Date: 2021-12-13
+- Reference Issues:
+  - Previous: https://github.com/withastro/rfcs/pull/44
+- Implementation PR: https://github.com/withastro/astro/pull/2168
+
+# Summary
+
+In order to improve build performance and scale to sites that build thousands of pages, deprecate dynamic features that forces our build to scan rendered HTML and provide static replacements.
+
+These features will be required for SSR support as well, so this RFC seeks to unify how Astro sites are built in a way that is workable both for static-site generation (SSG) and server-side rendering (SSR).
+
+# Example
+
+The following features are bundled as part of this RFC:
+
+- Enforced static use of `client:` directives.
+- Deprecation of `Astro.resolve`.
+- Introduction of `local:` directive to act as a static replacement for Astro.resolve.
```

Comment:
Have you checked with @jonathantneal that he's +1 on merging that idea with this PR? I know he did a lot of thinking in that original RFC and I'm sure would have good thoughts/feedback on this.

Reply:
No, I didn't know the status of that RFC. I think the deprecations in this RFC are an easier sell when including this, that's why I added it.

Reply:
We should talk about this because I can see 3 options here:

1. Bring `local:` into this RFC.
1. Make `local:` its own RFC and have that block this one.
1. Remove `local:` from this RFC and go ahead with it.

A little concerned that __(3)__ puts users into an uncommon workflow (using JavaScript imports).

Reply:
Because this is being adopted via an opt-in flag, I don't think the risk of 3 is too high while we're in the opt-in period. I agree that shipping it on for everyone without something _like_ `local:X` is a little more tricky and I'd share the same concern.

If you think that this RFC can push for `local:` without being a distraction the, that's still fine by me! Just making sure that in that case, you and @jonathantneal have a chance to sync anytime before the RFC call since I can't imagine that he wouldn't have valuable feedback.